src/main/java/com/tamaized/voidfog/FogRenderer.java

```diff
@@ -46,10 +46,26 @@ public void render(Camera camera,  FogType type, float viewDistance, boolean thi
 
         distance = MathHelper.lerp(delta / (distance > lastFogDistance ? 20 : 2), lastFogDistance, distance);
 
-        lastFogDistance = distance;
+        
+        //MathHelper.lerp(entityDelta,RenderSystem.getShaderFogStart(),getFogStart(distance, type, world, thickFog))
 
-        RenderSystem.setShaderFogStart(Math.min(RenderSystem.getShaderFogStart(), getFogStart(distance, type, world, thickFog)));
-        RenderSystem.setShaderFogEnd(Math.min(RenderSystem.getShaderFogEnd(), getFogEnd(distance, type, world, thickFog)));
+        lastFogDistance = distance;
+        if(((float)(entity.getY()-world.getBottomY()) <= ((float)(VoidFog.config.maxFogHeight)))||VoidFog.config.fancierFog) {
+            RenderSystem.setShaderFogStart(getFogStart(distance, type, world, thickFog));
+            RenderSystem.setShaderFogEnd(getFogEnd(distance, type, world, thickFog));
+        }
+        else if(((float)(entity.getY()-world.getBottomY()) <= ((float)(VoidFog.config.maxFogHeight))+VoidFog.config.fadeStartHeight)&&!VoidFog.config.fancierFog) {
+            //float entityAltitude = ((float)(entity.getY()-world.getBottomY()));
+            //float end = VoidFog.config.maxFogHeight;
+            //float start = (VoidFog.config.maxFogHeight+15F);
+            //float entityDelta = 1-(((float)(entity.getY()-world.getBottomY()))-VoidFog.config.maxFogHeight)/15F;
+            RenderSystem.setShaderFogStart(MathHelper.lerp(1-(((float)(entity.getY()-world.getBottomY()))-VoidFog.config.maxFogHeight)/VoidFog.config.fadeStartHeight,RenderSystem.getShaderFogStart(),getFogStart(distance, type, world, thickFog)));
+            RenderSystem.setShaderFogEnd(MathHelper.lerp(1-(((float)(entity.getY()-world.getBottomY()))-VoidFog.config.maxFogHeight)/VoidFog.config.fadeStartHeight,RenderSystem.getShaderFogEnd(),getFogEnd(distance, type, world, thickFog)));
+        }
+        else {
+            RenderSystem.setShaderFogStart(RenderSystem.getShaderFogStart());
+            RenderSystem.setShaderFogEnd(RenderSystem.getShaderFogEnd());
+        }
```

Comment:
Can you explain what all of these changes are doing?

Also please tidy up the formatting.

Reply:
This was my attempt to fix #11 

First if checks to see if the player is below the void fog height or if fancier fog is enabled. If so, render the denser fog as mentioned in #11.
First else if checks to see if the player is within VoidFog.config.fadeStartHeight blocks of the void fog height. If so, begin lerping fog from vanilla fog to denser fog based on player height. (default value is 15 blocks. Didn't make it a slider because I wasn't sure how to implement it in a way that makes sense. VoidFog.config.fadeStartHeight is additive to VoidFog.config.maxFogHeight, not an absolute position)
Catchall else simply renders vanilla fog.

Reply:
Tidied up formatting in [7172273](https://github.com/Sollace/Void-Fog/pull/13/commits/71722730d9f5249fb12f839af0cba18aadfe904c), [0fb2078](https://github.com/Sollace/Void-Fog/pull/13/commits/0fb20785cb9b52e2be07b3ae03fcbcc487abea7f), and [8d51fb5](https://github.com/Sollace/Void-Fog/pull/13/commits/8d51fb5b933e07e430c4f3df292aff3740ff2429). I renamed fadeStartHeight to fadeStartOffset for clarity.

entityDelta is converting entityAltitude into a percentage (0.0-1.0 = 0% - 100%) of the range of VoidFog.config.maxFogHeight to fadeStart (maxFogHeight+fadeStartOffset), which is then used in the MathHelper.lerp line to determine how to transition from vanilla fog to the denser void fog. Because I'm only using MathHelper.lerp when the player is within that range, there should be a relatively smooth transition as the player approaches the void fog height. I used a default of 15 for fadeStartOffset, just to start with. Haven't really played with different numbers to see what looks best. Offset can technically be set to 0 to "disable" the fade. I might be able to turn that into an option - I should also probably do some checks to prevent negative numbers on the offset.

Reply:
Added a check in [8dfd691](https://github.com/Sollace/Void-Fog/pull/13/commits/8dfd691d932c69278a9e985af9a81b14722d0278)